src/dune_rules/exe.ml

```diff
@@ -81,7 +81,10 @@ module Linkage = struct
           | Native -> Native
           | Best ->
             if Result.is_ok ctx.ocamlopt then Native
-            else Byte_with_stubs_statically_linked_in)
+            else Byte_with_stubs_statically_linked_in
+          | Melange ->
+            User_error.raise
+              [ Pp.textf "Linking is not supported in melange mode" ])
```

Comment:
The record inside `Other` should use the old `mode` to avoid this check if possible.

Reply:
This was fixed in https://github.com/jchavarri/dune/commit/e8c901c43eb07c4931dbc42a587d18f64388b182 (see [related discussion](https://github.com/jchavarri/dune/pull/4#discussion_r992833264))